Riot/Modules/Communities/Home/GroupHomeViewController.m

```diff
@@ -497,39 +498,73 @@ - (void)refreshGroupLongDescription
                                                  ];
         
         // Do some sanitisation by handling the potential image
+        MXWeakify(self);
         sanitisedGroupLongDescription = [MXKTools sanitiseHTML:_group.summary.profile.longDescription withAllowedHTMLTags:allowedHTMLTags imageHandler:^NSString *(NSString *sourceURL, CGFloat width, CGFloat height) {
             
-            NSString *imageURL;
+            NSString *localSourcePath;
             
             if (width != -1 && height != -1)
             {
                 CGSize size = CGSizeMake(width, height);
-                imageURL = [self.mxSession.matrixRestClient urlOfContentThumbnail:sourceURL toFitViewSize:size withMethod:MXThumbnailingMethodScale];
+                // Build the cache path for the a thumbnail of this image.
+                NSString *cacheFilePath = [MXMediaManager thumbnailCachePathForMatrixContentURI:sourceURL
+                                                                              andType:nil
+                                                                             inFolder:kMXMediaManagerDefaultCacheFolder
+                                                                        toFitViewSize:size
+                                                                           withMethod:MXThumbnailingMethodScale];
+                // Check whether the provided URL is a valid Matrix Content URI.
+                if (cacheFilePath)
+                {
+                    localSourcePath = [NSString stringWithFormat:@"file://%@", cacheFilePath];
+                    
+                    // Download the thumbnail if it is not already stored in the cache.
+                    if (![[NSFileManager defaultManager] fileExistsAtPath:cacheFilePath])
+                    {
+                        MXStrongifyAndReturnValueIfNil(self, nil);
```

Comment:
Move `MXStrongifyAndReturnValueIfNil(self, nil);` at the beginning of the block.